onnx/__init__.py

```diff
@@ -47,13 +49,16 @@ def _serialize(proto):
     if isinstance(proto, bytes):
         return proto
     elif hasattr(proto, 'SerializeToString') and callable(proto.SerializeToString):
-        return proto.SerializeToString()
+        result = proto.SerializeToString()  # type: bytes
+        return result
     else:
         raise ValueError('No SerializeToString method is detected. '
                          'neither proto is a str.\ntype is {}'.format(type(proto)))
 
 
-def _deserialize(s, proto):
+_Proto = TypeVar('_Proto', bound=google.protobuf.message.Message)
```

Comment:
Move this above so `_serialize` can also use it?

Reply:
Not needed above. The reason for this is that _Proto makes _deserialize() a generic function. If you call it with ModelProto, it returns a ModelProto. If you call it with NodeProto, it returns a NodeProto. _serialize() above only uses the proto parmeter once, so default variance is doing the job.